apisix/core/utils.lua

```diff
@@ -112,13 +104,13 @@ local function dns_parse(domain, resolvers)
         return nil, "unsupport DNS answer"
     end
 
-    return dns_parse(answer.cname, resolvers)
+    return dns_parse(answer.cname)
 end
 _M.dns_parse = dns_parse
 
-
-function _M.set_resolver(resolvers)
-    _M.resolvers = resolvers
```

Comment:
To maintain the backward compatibility, we still need to set `_M.resolvers`. The API in `core` is expected unchanged.

Reply:
I add this method and to do init logical to keep when set resolver the dns server should init again.

Reply:
After check their is no place use this method, so we can remove it.